rllib/algorithms/algorithm_config.py

```diff
@@ -1062,16 +1250,22 @@ def multi_agent(
     ) -> "AlgorithmConfig":
         """Sets the config's multi-agent settings.
 
+        Validates the new multi-agent settings and translates everything into
+        a unified multi-agent setup format. For example a `policies` list or set
```

Comment:
Doing most of the proper "translation" of the multi-agent config here helps us avoid having to do this inside e.g. RolloutWorker (when config should already have been frozen).